File: go/stocks/portfolio.go
package stocks

type Portfolio []Money

func (p Portfolio) Add(money Money) Portfolio {
	p = append(p, money)
	return p
}

func (p Portfolio) Evaluate(currency string) Money {
	total := 0.0
	for _, m := range p {
		total = total + convert(m, currency)
	}
	return Money{amount: total, currency: currency}
}

func convert(m Money, currency string) float64 {
	eurToUsd := 1.2
	if m.currency == currency {
		return m.amount
	}

	return m.amount * eurToUsd
}

File: typescript/src/portfolio.ts
import Money from "./money";


export class Portfolio {
    moneys: Money[] = [];

    add(money: Money): Portfolio {
        this.moneys.push(money);
        return this;
    }

    evaluate(currency: string): Money {
        const total = this.moneys.reduce((sum, money) => sum + this.convert(money, currency), 0);

        return new Money(total, currency);
    }

    convert(money: Money, currency: string): number {
        const exchangeRates = new Map();
        exchangeRates.set("EUR->USD", 1.2);
        exchangeRates.set("USD->KRW", 1100);

        if (money.currency === currency) {
            return money.amount;
        }

        const key = `${money.currency}->${currency}`
        const rate = exchangeRates.get(key) || 0;

        return money.amount * rate;
    }
}

export default Portfolio;